x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/Driver.java

```diff
@@ -305,6 +310,18 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * Releases all blocks in the given Page - ignoring and swallowing exceptions. This is a safety
+     * net used in failure scenarios - pages may or may not already be closed, but we need to be sure.
+     */
+    static void releasePageBlocksWhileHandlingException(Page page) {
+        for (int i = 0; i < page.getBlockCount(); i++) {
+            try {
+                Releasables.closeWhileHandlingException(page.getBlock(i));
+            } catch (IllegalStateException ignoreAlreadyClosed) {}
+        }
```

Comment:
Could you push it to `Page`? I suppose you don't even need to `catch` in that case - right, it can check if the blocks are closed?